### DISCUSSION TITLE
Device Type is based on screen-size rather than userAgent

### DISCUSSION THREAD

**Discussion by ACPK** [plausible/analytics#1064] [Anything Else]
I saw somewhere (cannot find the Github _issue_) that device types are based not the screen-size rather than the actual device type (i.e., _userAgent_). 

**Cause**: Julie, one of our users, is loading our tool in an iFrame on a desktop computer.  All of our users use our tool inside of an iFrame.

**Effect**: Our Plausible Analytics dashboard shows that 100% of our users are on mobile devices.

**Potential Solution:**

1. Switch the device type to be based on the _userAgent_.
2. Write a script that changes all places the "device type" was saved Clickhouse to be based on the _userAgent_.
3. **Bonus**: Include information on how self-hosted users can run the script in the _change log_.

**Comment by ACPK** [comment]
I'm curious on why this was moved to discussions? The dashboard is incorrectly telling me that the visitors were on mobile devices, and we wouldn't want to share the wrong info with clients.

**Reply by metmarkosaric** [reply]
Thanks! I moved it here as we keep "Issues" to actual bugs. In issue template, we kindly ask you to submit any feedback or requests here instead. We also kindly ask you to use the search function before submitting in order not to post duplicates. There's another thread with more info on this #157 

This report works as intended and [as documented](https://plausible.io/docs/devices). This report was built this way to inform better development decisions. When designing a website, what matters is how people view the website. The total available screen real estate is irrelevant when it comes to css media queries.

**Comment by ukutaht** [comment]
This change is on the roadmap: https://github.com/plausible/analytics/projects/1

Unfortunately it's not possible to make that change work retroactively. We don't store raw user-agent values.

**Comment by metmarkosaric** [comment]
**Update:** This is now live! From today, the devices are categorized into desktop, mobile or tablet. This is derived from the User-Agent HTTP header. The full User-Agent is discarded.

**Reply by dinandmentink** [reply]
Just making a note here that the Screen Size portion of the metrics documentation still states:

> Shows the width of the screens used by your visitors. We measure the width of the browser window where your site is actually rendered rather than the full screen width. Anything under 576px screen size is considered a mobile device, up to 992px is considered a tablet, up to 1440px is considered a laptop and anything above 1440px is considered a desktop. Metrics are shown as the total number of unique visitors and as a percentage of all visitors.

https://plausible.io/docs/metrics-definitions#screen-size

**Reply by metmarkosaric** [reply]
thanks for the heads up @dinandmentink! fixed now and it'll go live on site next time the cache clears
